src/components/BodyPortal.tsx

```diff
@@ -23,20 +23,25 @@ const getInsertBeforeTarget = (bodyPortalSlots: string[], slot?: string) => {
 
 export type BodyPortalProps = React.PropsWithChildren<{
   className?: string;
+  portal?: boolean;
   role?: string;
   slot?: string;
-  tagName?: string;
+  // List only elements that can have children
+  tagName?: 'div' | 'span' | 'nav' | 'main' | 'header' | 'footer';
   id?: string;
   'data-testid'?: string;
 }>;
 
 export const BodyPortal = React.forwardRef<HTMLElement, BodyPortalProps>((
-  { children, className, role, slot, tagName, id, ...props }, ref?: React.ForwardedRef<HTMLElement>
+  { children, className, portal = true, role, slot, tagName, id, ...props }, ref?: React.ForwardedRef<HTMLElement>
 ) => {
-  const tag = tagName?.toUpperCase() ?? 'DIV';
-  const internalRef = React.useRef<HTMLElement>(document.createElement(tag));
-  if (internalRef.current.tagName !== tag) {
-    internalRef.current = document.createElement(tag);
+  const TagName = tagName ?? 'div';
+  if (!portal) { return <TagName className={className} id={id} role={role} {...props}>{children}</TagName>; }
```

Comment:
Because StyledToastContainer subclasses BodyPortal, it seemed easier to do it this way rather than dynamically changing which class it subclasses.